common/src/main/java/org/axonframework/common/ReflectionUtils.java

```diff
@@ -584,6 +589,99 @@ public static String fieldNameFromMember(@Nonnull Member member) {
         }
     }
 
+    /**
+     * Each property of a record has both a backing {@link Field} and a {@link Method} (the accessor). This method
+     * filters out the fields that have a corresponding method, as these would result in duplicate child entity
+     * otherwise.
+     *
+     * @param methods The list of methods to check against the fields.
+     * @return <code>true</code> if the field is not duplicated by a given method
+     */
+    private static Predicate<Field> deduplicateRecordFields(List<Method> methods) {
+        return field -> methods
+                             .stream()
+                             .noneMatch(method -> method.getName().equals(field.getName())
+                                     && method.getParameterCount() == 0
+                                     && method.getReturnType().equals(field.getType()));
+    }
+
+    /**
+     * Collects all {@link Member}s of type {@link Field} and {@link Method} for given type in one single list.
+     * If the given type is a record, the record-style getters are deduplicated.
+     *
+     * @param type the type which fields and methods are collectd
+     * @return list of all found fields and messages
+     * @see #collectAnnotatedMethodsAndFields(Class, Predicate)
+     */
+    public static List<? extends Member> collectAnnotatedMethodsAndFields(Class<?> type) {
+        return collectAnnotatedMethodsAndFields(type, it -> true);
+    }
+
+    /**
+     * Collects all {@link Member}s of type {@link Field} and {@link Method} which match the given filter
```

Comment:
Writing the JavaDoc like so makes "members" the clickable link. Otherwise, it's "member"s, where the 's' wouldn't be part of the link:
```suggestion
     * Collects all {@link Member Members} of type {@link Field} and {@link Method} which match the given filter
```